compiler/semtypes.nim

```diff
@@ -1297,7 +1297,7 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
         let param = strTableGet(c.signatures, arg.name)
         if param != nil: typ = param.typ
         else:
-          localError(c.config, a.info, "typeless parameters are obsolete")
+          localError(c.config, a.info, "typeless parameters are not allowed, except for 'template'; See 'using' statement instead")
```

Comment:
```suggestion
          localError(c.config, a.info, "typeless parameters are not allowed, except with 'using' or for 'template|macro|generic' parameters")
```

how about this?

Reply:
Wait, AFAIK Generic has types, are generic but still types ?. :thinking:

Reply:
```nim
proc fn[T]() # typeless param
proc fn[T: enum]() # param with type class specification
```
that's what i mean; but maybe ok to remove `|generic` if unclear

Reply:
Oh ok. :+1: